Sources/ClientRuntime/Networking/Http/URLSession/FoundationStreamBridge.swift

```diff
@@ -182,20 +157,22 @@ class FoundationStreamBridge: NSObject, StreamDelegate {
         }
     }
 
-    private class WriteToOutputStreamResult: NSObject {
-        var data = Data()
-        var error: Error?
-    }
-
     /// Write the passed data to the output stream, using the reserved thread.
     private func writeToOutputStream(data: Data) async throws {
         try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
-            Self.queue.async {
-                let result = WriteToOutputStreamResult()
-                result.data = data
-                let selector = #selector(self.writeToOutputStreamOnThread)
-                self.perform(selector, on: Self.thread, with: result, waitUntilDone: true)
-                if let error = result.error {
+            queue.async { [self] in
+                guard !buffer.isEmpty || !data.isEmpty else { continuation.resume(); return }
+                buffer.append(data)
+                var writeCount = 0
+                buffer.withUnsafeBytes { bufferPtr in
+                    let bytePtr = bufferPtr.bindMemory(to: UInt8.self).baseAddress!
```

Comment:
Could use `if let bytePtr = bufferPtr.bindMemory(to: UInt8.self).baseAddress` to remove bang `!`